src/main/java/killrvideo/configuration/DseConfiguration.java

```diff
@@ -129,9 +145,59 @@ private void populateContactPoints(Builder clusterConfig)  {
                     .collect(Collectors.toList());            
         clusterConfig.withPort(clusterNodeAdresses.get(0).getPort());
         clusterNodeAdresses.stream()
-                           .map(adress -> adress.getHostName())
+                           .map(address -> address.getHostName())
                            .forEach(clusterConfig::addContactPoint);
     }
+
+    /**
+     * If SSL is enabled use the supplied CA cert file to create
+     * an SSL context and use to configure our cluster.
+     *
+     * @param clusterConfig
+     *      current configuration
+     */
+    private void populateSSL(Builder clusterConfig) {
+        if (dseEnableSSL.isPresent() &&  dseEnableSSL.get()) {
```

Comment:
Idea: `if (dseEnableSSL.isPresent() &&  dseEnableSSL.get()) {` could be directly moved in caller do not invoke populateSSL at all if the flag is not enabled.

Reply:
Yea, makes sense

Reply:
Are you thinking the same thing with the populateAuthentication() method then as well?  Otherwise we'll essentially have 2 different implementation styles in place.

Reply:
Yes makes sense to do it with `populateAuthentication()` as well, sure. 👍will do when merged

Reply:
Ok, taking a look now.

Reply:
Ok, did some refactoring here.